### PR TITLE
Add virtual relationship check

### PR DESCRIPTION
**Pull request by JMHReif**
Signed-off-by: Florent Biville <florent.biville@neo4j.com>

### CODE REVIEW

**Comment by JMHReif** [comment]
@conker84, should we update the equalsHashCode of virtual nodes and virtual relationships? Does the compare work when dealing with virtual nodes/rels?

**Comment by conker84** [comment]
@JMHReif @fbiville I have a proposal for this, why we don't get rid of VirtualPath and use the `PathImpl` class? So we can delegate any kind of business logic to it? I did a quick research and seems that `VirtualPath` is used only one point so the change should be very quick

File: core/src/main/java/apoc/result/VirtualPath.java [hunk 1]
```diff
@@ -113,9 +113,30 @@ public void remove() {
...
+		Node previousStartNode;
+		Node previousEndNode;
+		Relationship previousRelationship = lastRelationship();
+		if (previousRelationship != null) {
+			previousStartNode = previousRelationship.getStartNode();
+			previousEndNode = previousRelationship.getEndNode();
+		} else {
+			previousStartNode = startNode();
+			previousEndNode = endNode();
+		}
```
Comment (conker84) [inline core/src/main/java/apoc/result/VirtualPath.java:132]: Can you please fix the indentation?

**Review by conker84 — APPROVED** [review]

**Comment by conker84** [comment]
@JMHReif can you please rebase it

**Pull request merged**